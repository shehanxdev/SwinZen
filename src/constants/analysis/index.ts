export * from './analysis.enum';
export * from './pro-tips-content';
