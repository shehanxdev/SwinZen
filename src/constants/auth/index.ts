export * from './otp-data.enum';
