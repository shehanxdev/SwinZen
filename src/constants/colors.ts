/* eslint-disable @typescript-eslint/no-shadow */
/* eslint-disable @typescript-eslint/no-redeclare */

//NOTE:: Since an enum member cannot have a numeric name prefixed with "Sz"
enum Primary {
  Sz900 = `#070707`,
  Sz800 = `#0C1E0F`,
  Sz700 = `#16501D`,
  Sz600 = `#28642A`,
  Sz500 = `#45734A`,
  Sz400 = `#69AA67`,
  Sz300 = `#8DBE8B`,
  Sz200 = `#C0DBBF`,
  Sz100 = `#F0F7F0`, //NOTE::figma contain two different colors(#F0F7F0, #C0DBBF) with the border. #C0DBBF neeeds to be removed since it does not consider within the colors
}

enum Neutral {
  Sz1000 = `#283E36`, //NOTE::not available within the design system colors
  Sz900 = `#212529`,
  Sz800 = `#343A40`,
  Sz700 = `#495057`,
  Sz600 = `#6C757D`,
  Sz500 = `#ADB5BD`,
  Sz400 = `#CED4DA`,
  Sz300 = `#DEE2E6`,
  Sz200 = `#E9ECEF`,
  Sz100 = `#F8F9FA`, //NOTE::figma contain two different colors(#F8F9FA, #DEE2E6) with the border. #DEE2E6 neeeds to be removed since it does not consider within the colors
  Black = `#000000`,
  White = `#FFFFFF`,
}

enum Secondary {
  Sz900 = `#F65815`,
  Sz800 = `#F87944`,
  Sz700 = `#F99469`,
  Sz600 = `#FAA17C`,
  Sz500 = `#FBBCA1`,
  Sz400 = `#FDD7C7`,
  Sz300 = `#FDE4D9`,
  Sz200 = `#FFF2ED`,
  Sz100 = `#FFE3B0`, //NOTE::figma contain two different colors(##FFE3B0, #FFF9F6) with the border. #FFF9F6 neeeds to be removed since it does not consider within the colors
}

/* Currently this is available within the design system as a Secondary value. Since two different colors cannot exist in there, this has to be added to the design system */
enum Tertiary {
  Sz900 = `#A2FD2F`,
  Sz800 = `#B5FD59`,
  Sz700 = `#C4FE7A`,
  Sz600 = `#CBFE8A`,
  Sz500 = `#DAFEAC`,
  Sz400 = `#E9FFCD`,
  Sz300 = `#F0FEDE`,
  Sz200 = `#F8FFEF`,
  Sz100 = `#DAFEAC`, //NOTE::figma contain two different colors(#DAFEAC, #FBFFF7) with the border. #FBFFF7 neeeds to be removed since it does not consider within the colors
}

enum Transparency {
  Sz100 = `#212529`,
  Sz80 = `#212529cc`,
  Sz64 = `#212529a3`,
  Sz56 = `#2125298f`,
  Sz40 = `#21252966`,
  Sz24 = `#2125293d`,
  Sz16 = `#21252929`,
  Sz8 = `#21252914`,
  Sz4 = `#2125290a`,
  full = `#ffffff00`, //NOTE:: This is full transpapernt color. Not available under the design system.
}

enum Error {
  SzMain = `#E35F5F`, //Not defined within the design system but being used within the screens :(
}

const SwingZenThemePalette = {
  Primary,
  Neutral,
  Secondary,
  Tertiary,
  Transparency,
  Error,
};

export type Color = Primary | Neutral | Secondary | Tertiary | Transparency | Error;

export const Color = {
  ...SwingZenThemePalette,
};
