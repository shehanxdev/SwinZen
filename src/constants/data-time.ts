export const dateFormats = {
  default: 'YYYY-MM-DD',
  calenderHeader: 'MMMM YYYY',
  rangeDatePicker: 'MMM DD',
  scratchedDateAndTime: 'HH:MM, DD/MM/YY',
  singleDatePicker: 'ddd, MMM DD',
  textInput: 'DD / MM / YY',
  extrasMarketHeader: 'dd/MM/YY',
  finalResultsHeader: 'dd/MM/YYYY',
  dayAndDate: 'ddd, DD',
  timestamp: 'DD/MM/YYYY HH:mm:ssPM Z',
  advertisedStartTime: 'ddd, DD MMM YYYY HH:mm PM',
  saveFileTimestamp: 'DD_MM_YYYY_HH_mm_ss',
};
