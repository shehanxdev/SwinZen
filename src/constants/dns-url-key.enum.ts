//TODO::fill other dns url keys
export enum DnsUrlKey {
  Auth0 = 'auth0',
}
