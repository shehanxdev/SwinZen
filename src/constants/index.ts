export * from './auth';
export * from './error';
export * from './colors';
export * from './data-time';
export * from './navigation-routes';
export * from './regular-expressions';
export * from './runtime-environment';
export * from './static-info';
export * from './typography';
