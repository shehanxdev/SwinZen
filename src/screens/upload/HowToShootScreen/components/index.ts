export * from './HowToShootStepOne';
export * from './HowToShootStepTwo';
export * from './HowToShootStepThree';
export * from './HowToShootStepFour';
