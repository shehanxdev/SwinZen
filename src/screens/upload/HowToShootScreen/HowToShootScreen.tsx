import { useNavigation } from '@react-navigation/native';
import React, { useCallback, useEffect, useLayoutEffect, useMemo, useState } from 'react';
import { TouchableOpacity, View } from 'react-native';

import { Button, CrossIcon, Link, Text } from '@sz/components';
import { tw } from '@sz/config';
import { Color, Route, SetupValuesType, TextAlignment, TextVariant } from '@sz/constants';
import { VideoSetupValuesType } from '@sz/models';
import { NavigationService } from '@sz/services';
import { useDispatch, useSelector } from '@sz/stores';

import { BaseUploadScreen, InstructionsSkipModal } from '../components';

const TEST_ID_PREFIX = 'HowToShootScreenTestID';

export function HowToShootScreen({ route }) {
  const data = route.params.params?.setupValues as VideoSetupValuesType;

  const [count, setCount] = useState(0);
  const [showModal, setShowModal] = useState(false);

  const navigation = useNavigation();
  const dispatch = useDispatch();

  const setSkippedInstructions = dispatch.persistentUserStore.setSkippedInstructions;
  const skippedInstructions = useSelector(state => state.persistentUserStore.skippedInstructions);

  const renderHeader = () => {
    return (
      <TouchableOpacity
        onPress={() => {
          setCount(0);
          NavigationService.goBack();
        }}
        style={tw`py-2.5 pr-2.5`}>
        <CrossIcon />
      </TouchableOpacity>
    );
  };

  // custom action for navigation header back button
  useLayoutEffect(() => {
    navigation.setOptions({
      headerRight: renderHeader,
    });
  });

  useEffect(() => {
    if (data) {
      if (data.videoView === SetupValuesType.DOWN_THE_LINE) {
        if (data.shootingMethod === SetupValuesType.HAND_HELD) {
          setCount(1);
        }
      } else if (data.videoView === SetupValuesType.FACE_ON) {
        if (data.shootingMethod === SetupValuesType.TRIPOD) {
          setCount(2);
        } else {
          setCount(3);
        }
      }
    }
  }, []);

  const onSkipPress = () => {
    setCount(0);
    setShowModal(false);
    setSkippedInstructions(true);
    setTimeout(() => NavigationService.navigate(Route.PreValidation), 200);
  };

  const StepOneContent = useCallback(() => {
    return (
      <View>
        <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Down The Line/Tripod Instructions\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Take 4 paces/12 ft behind the ball pointing towards the target.\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Set your tripod 4-5 feet high.\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Position your phone in portrait mode.\n`}
        </Text>
      </View>
    );
  }, []);

  const StepTwoContent = useCallback(() => {
    return (
      <View>
        <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Down The Line/Hand-Held Instructions\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Have your buddy take 4 paces/12 ft behind the ball pointing towards the target.\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Have your buddy hold the phone in portrait mode.\n`}
        </Text>
      </View>
    );
  }, []);

  const StepThreeContent = useCallback(() => {
    return (
      <View>
        <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Face-On/Tripod Instructions\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Take 3 paces/9 ft from the ball, directly in front of the golfer.\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Set your tripod 4-5 feet high.\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Position your phone in landscape mode.\n`}
        </Text>
      </View>
    );
  }, []);

  const StepFourContent = useCallback(() => {
    return (
      <View>
        <Text color={Color.Primary.Sz200} variant={TextVariant.Body1SemiBold} textAlign={TextAlignment.Auto}>
          {`Face-On/Hand-Held Instructions\n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Have your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. \n`}
        </Text>
        <Text color={Color.Neutral.White} variant={TextVariant.Body1Regular} textAlign={TextAlignment.Auto}>
          {`Have your buddy take 3 paces/9 ft from the ball, directly in front of the golfer. \n`}
        </Text>
      </View>
    );
  }, []);

  const renderBody = useMemo(() => {
    switch (count) {
      case 0:
        return <StepOneContent />;
      case 1:
        return <StepTwoContent />;
      case 2:
        return <StepThreeContent />;
      case 3:
        return <StepFourContent />;
      default:
        return null;
    }
  }, [count]);

  return (
    <BaseUploadScreen testID={TEST_ID_PREFIX}>
      <View style={tw`flex-1 justify-between mx-4 mt-8`}>
        {renderBody}
        <View style={tw`mt-25 mb-3`}>
          <Button
            onPress={() => {
              if (count === 3 && !data) {
                NavigationService.goBack();
              } else if (data) {
                setCount(0);
                NavigationService.navigate(Route.PreValidation);
              } else {
                setCount(count + 1);
              }
            }}
            title="Next"
          />
          {!skippedInstructions && data && (
            <View style={tw`my-6`}>
              <Link text="Always skip instructions" onPress={() => setShowModal(true)} />
            </View>
          )}
        </View>
      </View>
      <InstructionsSkipModal
        showModal={showModal}
        handleModalClose={() => setShowModal(false)}
        onSkipped={onSkipPress}
      />
    </BaseUploadScreen>
  );
}
