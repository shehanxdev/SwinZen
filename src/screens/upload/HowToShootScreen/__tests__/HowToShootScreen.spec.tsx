import * as React from 'react';

import { SetupValuesType } from '@sz/constants';
import { renderWithProviders } from '@sz/test-utils';

import { HowToShootScreen } from '../HowToShootScreen';

describe('HowToShoot Screen screen', () => {
  const testID = 'HowToShootScreenTestID';
  const mockRoute = {
    params: {
      params: {
        setupValues: {
          videoView: SetupValuesType.DOWN_THE_LINE,
          shootingMethod: SetupValuesType.HAND_HELD,
        },
      },
    },
  };
  const getRenderedScreen = () => renderWithProviders(<HowToShootScreen route={mockRoute} />);

  it(`should render HowToShoot Screen correctly`, async () => {
    const renderer = getRenderedScreen();
    const renderTree = renderer.toJSON();
    expect(renderTree).toMatchSnapshot();
  });

  it('should find the HowToShoot Screen via testID', () => {
    const { getByTestId } = getRenderedScreen();
    const foundScreen = getByTestId(testID);
    expect(foundScreen).toBeDefined();
  });
});
