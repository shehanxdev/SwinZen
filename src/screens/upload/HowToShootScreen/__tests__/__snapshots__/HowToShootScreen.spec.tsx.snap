// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HowToShoot Screen screen should render HowToShoot Screen correctly 1`] = `
<View
  collapsable={false}
  pointerEvents="box-none"
  style={
    {
      "flex": 1,
    }
  }
>
  <View
    accessibilityIgnoresInvertColors={true}
    style={
      {
        "backgroundColor": "#212529",
        "height": "100%",
        "paddingTop": 50,
      }
    }
  >
    <Image
      style={
        [
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
          {
            "height": "100%",
            "width": undefined,
          },
          undefined,
        ]
      }
      testID="HowToShootScreenTestID"
    />
    <RCTSafeAreaView
      style={
        {
          "height": "100%",
          "paddingTop": 50,
        }
      }
    >
      <RCTScrollView
        contentContainerStyle={
          {
            "flexGrow": 1,
          }
        }
      >
        <View>
          <View
            style={
              {
                "flexBasis": "0%",
                "flexGrow": 1,
                "flexShrink": 1,
                "justifyContent": "space-between",
                "marginLeft": 16,
                "marginRight": 16,
                "marginTop": 32,
              }
            }
          >
            <View>
              <Text
                style={
                  [
                    {
                      "color": "#C0DBBF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-SemiBold",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Down The Line/Hand-Held Instructions

              </Text>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                Have your buddy take 4 paces/12 ft behind the ball pointing towards the target.

              </Text>
              <View
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderColor": "#FFFFFF",
                      "borderWidth": 4,
                      "height": 206,
                      "width": "100%",
                    },
                  ]
                }
              >
                <FastImageView
                  defaultSource={null}
                  resizeMode="cover"
                  source={
                    {
                      "cache": "immutable",
                      "priority": "high",
                      "uri": "https://video-thumbnails.dev.swingzen.com/Take%20Down%20the%20Line%204%20paces%20Behind%20Ball.gif",
                    }
                  }
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#FFFFFF",
                      "lineHeight": 28,
                    },
                    {
                      "fontFamily": "SourceSansPro-Regular",
                      "fontSize": 18,
                      "fontWeight": "normal",
                      "textAlign": "auto",
                    },
                  ]
                }
              >
                
Have your buddy hold the phone in portrait mode.

              </Text>
              <View
                style={
                  [
                    {
                      "overflow": "hidden",
                    },
                    {
                      "borderColor": "#FFFFFF",
                      "borderWidth": 4,
                      "height": 206,
                      "width": "100%",
                    },
                  ]
                }
              >
                <FastImageView
                  defaultSource={null}
                  resizeMode="cover"
                  source={
                    {
                      "cache": "immutable",
                      "priority": "high",
                      "uri": "https://video-thumbnails.dev.swingzen.com/Landscape%20and%20Portrait.gif",
                    }
                  }
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                />
              </View>
            </View>
            <View
              style={
                {
                  "marginBottom": 12,
                  "marginTop": 100,
                }
              }
            >
              <View
                style={
                  {
                    "flexDirection": "row",
                  }
                }
              >
                <View
                  collapsable={false}
                  onTouchEnd={[Function]}
                  onTouchStart={[Function]}
                  style={
                    {
                      "backgroundColor": "#A2FD2F",
                      "borderColor": "#00000000",
                      "borderRadius": 10,
                      "borderStyle": "solid",
                      "borderWidth": 1,
                      "elevation": 0,
                      "justifyContent": "center",
                      "minWidth": "auto",
                      "shadowColor": "#000",
                      "shadowOffset": {
                        "height": 0,
                        "width": 0,
                      },
                      "shadowOpacity": 0,
                      "shadowRadius": 0,
                      "width": "100%",
                    }
                  }
                >
                  <View
                    accessibilityRole="button"
                    accessibilityState={
                      {
                        "disabled": false,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      [
                        {
                          "overflow": "hidden",
                        },
                        {
                          "borderRadius": 10,
                        },
                      ]
                    }
                  >
                    <View
                      style={
                        [
                          {
                            "alignItems": "center",
                            "flexDirection": "row",
                            "justifyContent": "center",
                          },
                          {
                            "height": 48,
                          },
                        ]
                      }
                    >
                      <Text
                        numberOfLines={1}
                        selectable={false}
                        style={
                          [
                            {
                              "color": "#000000",
                              "fontFamily": "SourceSansPro-Regular",
                              "fontWeight": "normal",
                            },
                            {
                              "textAlign": "left",
                            },
                            [
                              {
                                "letterSpacing": 1,
                                "marginHorizontal": 16,
                                "marginVertical": 9,
                                "textAlign": "center",
                              },
                              {
                                "marginHorizontal": 8,
                              },
                              {
                                "textTransform": "uppercase",
                              },
                              {
                                "color": "#6200ee",
                                "fontFamily": "SourceSansPro-SemiBold",
                                "fontWeight": "normal",
                              },
                              {
                                "fontFamily": "SourceSansPro-SemiBold",
                                "fontWeight": "normal",
                              },
                              {
                                "color": "#283E36",
                              },
                            ],
                          ]
                        }
                      >
                        <Text
                          style={
                            [
                              {
                                "color": "#283E36",
                                "lineHeight": 28,
                              },
                              {
                                "fontFamily": "SourceSansPro-SemiBold",
                                "fontSize": 18,
                                "fontWeight": "bold",
                                "textAlign": "center",
                              },
                            ]
                          }
                        >
                          Next
                        </Text>
                      </Text>
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  {
                    "marginBottom": 24,
                    "marginTop": 24,
                  }
                }
              >
                <Text
                  onPress={[Function]}
                  style={
                    [
                      {
                        "color": "#A2FD2F",
                        "lineHeight": 20,
                      },
                      {
                        "fontFamily": "SourceSansPro-Regular",
                        "fontSize": 16,
                        "fontWeight": "bold",
                        "textAlign": "center",
                        "textDecorationLine": "underline",
                      },
                    ]
                  }
                >
                  Always skip instructions
                </Text>
              </View>
            </View>
          </View>
          <View
            testID="InstructionsSkipModalTestID"
          >
            <Modal
              animationType="none"
              deviceHeight={null}
              deviceWidth={null}
              hardwareAccelerated={false}
              hideModalContentWhileAnimating={false}
              onBackdropPress={[Function]}
              onModalHide={[Function]}
              onModalWillHide={[Function]}
              onModalWillShow={[Function]}
              onRequestClose={[Function]}
              panResponderThreshold={4}
              scrollHorizontal={false}
              scrollOffset={0}
              scrollOffsetMax={0}
              scrollTo={null}
              statusBarTranslucent={false}
              supportedOrientations={
                [
                  "portrait",
                  "landscape",
                ]
              }
              swipeThreshold={100}
              transparent={true}
              visible={false}
            />
          </View>
        </View>
      </RCTScrollView>
    </RCTSafeAreaView>
  </View>
</View>
`;
